### DISCUSSION TITLE
Add ```??:``` operator for ternary conditional operator

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5883] [Language Ideas]
so ```string SomeParameter = VarA == VarB ? VarNullableC??:string.Empty;``` Can work without writing ```&&VarC is not null```
of course the next line was ```SomeParameter!!;``` https://github.com/dotnet/csharplang/issues/2145

**Comment by TonyValenti** [comment]
Can you give an example
Of usage and what it means?

**Reply by Xyncgas** [reply]
I did give the example of usage, so VarNullableC which could be null is checked and if it's nulled then ```SomParameter=string.Empty;```
I came from a work place that I felt this was needed, but I can't convince you by going any further than telling you that I do think ```??:``` would be a nice thing to have and it's ok if I don't get it

**Reply by HaloFour** [reply]
This already works and is significantly clearer:

```cs
string SomeParameter = VarA == VarB && VarNullableC != null ? VarNullableC : string.Empty;
```

**Reply by Xyncgas** [reply]
but you would have to write ```VarNullableC != null```
that's why I am proposing the ```??:``` operator to make it more efficient
well whatever dude

**Reply by huoyaoyuan** [reply]
Code is read much more than written. An operator combining several operations is hard to read.

**Reply by GiottoVerducci** [reply]
You could write also ```string SomeParameter = VarA == VarB ? VarNullableC ?? string.Empty : string.Empty;```
or ```string SomeParameter = (VarA == VarB ? VarNullableC : null) ?? string.Empty;```

It's just a coincidence that the third expression of the ternary should be the same value as the coalesce value inside the second expression, it's really corner-case.

**Reply by Xyncgas** [reply]
I know, but you will have to repeat two things (string.Empty), when they meant the same thing
